File: P5 2024-2/Paradigmas de Programacao/08-30/POO/Retangulo.py
class Retangulo:

    def __init__(self, comprimento, largura):
        self.comprimento = comprimento
        self.largura = largura

    def calcular_area(self):
        return self.comprimento * self.largura
    
    def calcular_perimetro(self):
        return (2*self.comprimento)+(2*self.largura)

ret = Retangulo(200, 300)
print("Area: ", ret.calcular_area())
print("Perímetro: ", ret.calcular_perimetro())

File: P5 2024-2/Paradigmas de Programacao/08-30/POO/Retangulo.go
package main
import "fmt"

type Retangulo struct {
	comprimento, largura float64
}

func (r Retangulo) calcularArea() float64 {
	return r.comprimento * r.largura
}

func (r Retangulo) calcularPerimetro() float64 {
	return (2*r.comprimento)+(2*r.largura)
}

func main(){
	ret := Retangulo{200,300}
	fmt.Printf("Area: %.2f\n", ret.calcularArea())
	fmt.Printf("Perímetro: %.2f\n", ret.calcularPerimetro())
}
